Drivers/imu/imu.c

```diff
@@ -0,0 +1,151 @@
+/*
+    LSM6DSOXTR IMU DRIVER Source File
+    Link to part Datasheet for reference:
+    https://www.st.com/resource/en/datasheet/lsm6dsox.pdf
+
+    Author: Hamza Iqbal
+*/
+#include "imu.h"
+
+/* IMU Intitialization */
+uint8_t imu_init(IMU_DATA *imu, I2C_HandleTypeDef *i2cHandle)
```

Comment:
I think you forgot to rename some of your parameters, since it still says IMU_DATA. Also do i2c_handle